### ISSUE TITLE
Execution context is also lost with mercurius without enterWith: true

### ISSUE THREAD

**Issue by steinroe** [Papooch/nestjs-cls#1]
Hi,

unfortunately, Mercurius also requires `enterWith: true` to be true. The issue occurs not if using `fastify.inject` - only with "real clients". I have absolutely no idea why, but that seems to fix it. I opened a [PR](https://github.com/Papooch/nestjs-cls/pull/2) to replace `fastify.inject` with supertest in the mercurius e2e test and to set `useEnterWith: true`. I also updated the documentation accordingly.

**Comment by Papooch** [comment]
That's really unfortunate :( That means there's really no 100% safe way to use `cls` with GQL, because `AsyncLocalStorage.enterWith()` can leak context between concurrent calls.

Thanks for the PR, I have merged it, and I will publish it with a new version in a couple of days.

I'll have to do some more tests to verify that no context is leaked in a request after it goes past the `ClsMiddleware`, but I fear the worst. (I have discovered that I can access the context of the previous request before `enter` is called in the middleware)
